packages/super-components/components/msc-billing-tile/src/msc-billing-tile/msc-billing-contact.tsx

```diff
@@ -0,0 +1,197 @@
+
+  @Prop() localeStrings: Translations;
+
+  @State() domainProperties?: DomainProperties;
+
+  @State() ownerInfos?: ContactInfos;
+
+  async componentWillLoad() {
+    if (this.isDomainServiceType()) {
+      this.domainProperties = await apiClient.v6
+        .get(`/domain/${this.serviceName}`)
+        .then((response) => response.data);
+
+      this.ownerInfos = await apiClient.v6
+        .get(`/domain/contact/${this.domainProperties?.whoisOwner}`)
+        .then((response) => response.data);
+    }
+  }
+
+  private isDomainServiceType() {
+    return this.serviceType === 'DOMAIN';
+  }
+
+  private isOwnerChangeableServiceType() {
+    return [
+      'DOMAIN',
+      'HOSTING_WEB',
+      'HOSTING_PRIVATEDATABASE',
+      'DEDICATED_SERVER',
+      'DEDICATED_HOUSING',
+      'VPS',
+      'EMAIL_DOMAIN',
+      'VRACK',
+      'DEDICATEDCLOUD', // VMWare
+    ].includes(this.serviceType);
+  }
+
+  private getDomainOwnerFullname() {
+    return `${this.ownerInfos?.firstName} ${this.ownerInfos?.lastName}`;
+  }
+
+  private getMenuLinks() {
+    return (
+      <>
+        <osds-link
+          data-tracking={this.subscriptionManagementDataTracking}
+          href={`https://www.ovh.com/manager/#/dedicated/contacts/services?serviceName=${this.serviceName}`}
+          color={OdsThemeColorIntent.primary}
+          target={OdsHTMLAnchorElementTarget._blank}
+        >
+          {this.localeStrings.manager_billing_subscription_contacts_management}
+          <osds-icon
+            slot="end"
+            class="new-link-icon"
+            size={OdsIconSize.xxs}
+            name={OdsIconName.ARROW_RIGHT}
+            color={OdsThemeColorIntent.primary}
+          />
+        </osds-link>
+        {this.isOwnerChangeableServiceType() && (
+          <osds-link
+            data-tracking={this.changeOwnerDataTracking}
+            href={
+              this.isDomainServiceType()
+                ? `https://www.ovh.com/fr/order/domain/#/legacy/domain/trade/informations?options=~~(domain~~'${this.serviceName})`
+                : 'https://www.ovh.com/cgi-bin/fr/procedure/procedureChangeOwner.cgi'
+            }
+            target={OdsHTMLAnchorElementTarget._blank}
+            color={OdsThemeColorIntent.primary}
+          >
+            <span class="link-text">
+              {this.localeStrings.billing_services_actions_menu_change_owner}
+            </span>
+            <osds-icon
+              slot="end"
+              class="new-link-icon"
+              size={OdsIconSize.xxs}
+              name={OdsIconName.EXTERNAL_LINK}
+              color={OdsThemeColorIntent.primary}
+            />
+          </osds-link>
+        )}
+        {this.isDomainServiceType() && this.domainProperties?.whoisOwner && (
+          <osds-link
+            data-tracking={this.updateOwnerDataTracking}
+            href={`https://www.ovh.com/manager/#/dedicated/contact/${this.serviceName}/${this.domainProperties?.whoisOwner}`}
```

Comment:
**nitpick:** Use `navigation.getURL` function to build the URL

Reply:
We don't have access to navigation here
We have to create or duplicate a helper for this

Reply:
IMO, super-component can accept URL as input if we dont have access to `navigation` inside super-component.
Because, the current implementation only makes the supercomponent usable in EU and it doesn't work in CA, US and LABEU as the host name is different for each region.

Reply:
Can we just take the current hostname from `window.location.origin` ? Or is there limitations to this solution ?